[2024-07-18T16:02:27.927+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: google_analytics_data_etl.task_load_data_postgres manual__2024-07-18T15:04:45.851017+00:00 [queued]>
[2024-07-18T16:02:27.937+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: google_analytics_data_etl.task_load_data_postgres manual__2024-07-18T15:04:45.851017+00:00 [queued]>
[2024-07-18T16:02:27.938+0000] {taskinstance.py:2171} INFO - Starting attempt 6 of 7
[2024-07-18T16:02:27.960+0000] {taskinstance.py:2192} INFO - Executing <Task(PostgresOperator): task_load_data_postgres> on 2024-07-18 15:04:45.851017+00:00
[2024-07-18T16:02:27.970+0000] {standard_task_runner.py:60} INFO - Started process 1828 to run task
[2024-07-18T16:02:27.974+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'google_analytics_data_etl', 'task_load_data_postgres', 'manual__2024-07-18T15:04:45.851017+00:00', '--job-id', '190', '--raw', '--subdir', 'DAGS_FOLDER/google_analytics_data_etl/google_analytics_data_etl.py', '--cfg-path', '/tmp/tmpqea8can7']
[2024-07-18T16:02:27.976+0000] {standard_task_runner.py:88} INFO - Job 190: Subtask task_load_data_postgres
[2024-07-18T16:02:28.112+0000] {task_command.py:423} INFO - Running <TaskInstance: google_analytics_data_etl.task_load_data_postgres manual__2024-07-18T15:04:45.851017+00:00 [running]> on host 3d44d7374020
[2024-07-18T16:02:28.199+0000] {abstractoperator.py:708} ERROR - Exception rendering Jinja template for task 'task_load_data_postgres', field 'sql'. Template: "{{ ti.xcom_pull(task_ids=task_prepare_query_to_postgres, key='query_event_to_postgres') }}"
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/abstractoperator.py", line 700, in _do_render_template_fields
    rendered_content = self.render_template(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/template/templater.py", line 174, in render_template
    return self._render(template, context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/abstractoperator.py", line 654, in _render
    return super()._render(template, context, dag=dag)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/template/templater.py", line 131, in _render
    return render_template_to_string(template, context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/helpers.py", line 289, in render_template_to_string
    return render_template(template, cast(MutableMapping[str, Any], context), native=False)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/helpers.py", line 284, in render_template
    return "".join(nodes)
  File "<template>", line 13, in root
  File "/home/airflow/.local/lib/python3.8/site-packages/jinja2/sandbox.py", line 393, in call
    return __context.call(__obj, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/jinja2/runtime.py", line 298, in call
    return __obj(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 79, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3036, in xcom_pull
    query = XCom.get_many(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/api_internal/internal_api_call.py", line 114, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/xcom.py", line 518, in get_many
    query = query.filter(BaseXCom.task_id.in_(task_ids))
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 641, in in_
    return self.operate(in_op, other)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/orm/attributes.py", line 322, in operate
    return op(self.comparator, *other, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 1440, in in_op
    return a.in_(b)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 641, in in_
    return self.operate(in_op, other)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/orm/properties.py", line 426, in operate
    return op(self.__clause_element__(), *other, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 1440, in in_op
    return a.in_(b)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 641, in in_
    return self.operate(in_op, other)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 873, in operate
    return op(self.comparator, *other, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 1440, in in_op
    return a.in_(b)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 641, in in_
    return self.operate(in_op, other)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/type_api.py", line 77, in operate
    return o[0](self.expr, op, *(other + o[1:]), **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/default_comparator.py", line 159, in _in_impl
    seq_or_selectable = coercions.expect(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/coercions.py", line 193, in expect
    resolved = impl._literal_coercion(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/coercions.py", line 573, in _literal_coercion
    element = list(element)
  File "/home/airflow/.local/lib/python3.8/site-packages/jinja2/runtime.py", line 852, in _fail_with_undefined_error
    raise self._undefined_exception(self._undefined_message)
jinja2.exceptions.UndefinedError: 'task_prepare_query_to_postgres' is undefined
[2024-07-18T16:02:28.206+0000] {taskinstance.py:2699} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 2335, in _run_raw_task
    self._execute_task_with_callbacks(context, test_mode, session=session)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 2466, in _execute_task_with_callbacks
    task_orig = self.render_templates(context=context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 2878, in render_templates
    original_task.render_template_fields(context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/baseoperator.py", line 1248, in render_template_fields
    self._do_render_template_fields(self, self.template_fields, context, jinja_env, set())
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 79, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/abstractoperator.py", line 700, in _do_render_template_fields
    rendered_content = self.render_template(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/template/templater.py", line 174, in render_template
    return self._render(template, context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/abstractoperator.py", line 654, in _render
    return super()._render(template, context, dag=dag)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/template/templater.py", line 131, in _render
    return render_template_to_string(template, context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/helpers.py", line 289, in render_template_to_string
    return render_template(template, cast(MutableMapping[str, Any], context), native=False)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/helpers.py", line 284, in render_template
    return "".join(nodes)
  File "<template>", line 13, in root
  File "/home/airflow/.local/lib/python3.8/site-packages/jinja2/sandbox.py", line 393, in call
    return __context.call(__obj, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/jinja2/runtime.py", line 298, in call
    return __obj(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 79, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3036, in xcom_pull
    query = XCom.get_many(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/api_internal/internal_api_call.py", line 114, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/xcom.py", line 518, in get_many
    query = query.filter(BaseXCom.task_id.in_(task_ids))
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 641, in in_
    return self.operate(in_op, other)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/orm/attributes.py", line 322, in operate
    return op(self.comparator, *other, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 1440, in in_op
    return a.in_(b)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 641, in in_
    return self.operate(in_op, other)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/orm/properties.py", line 426, in operate
    return op(self.__clause_element__(), *other, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 1440, in in_op
    return a.in_(b)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 641, in in_
    return self.operate(in_op, other)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 873, in operate
    return op(self.comparator, *other, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 1440, in in_op
    return a.in_(b)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 641, in in_
    return self.operate(in_op, other)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/type_api.py", line 77, in operate
    return o[0](self.expr, op, *(other + o[1:]), **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/default_comparator.py", line 159, in _in_impl
    seq_or_selectable = coercions.expect(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/coercions.py", line 193, in expect
    resolved = impl._literal_coercion(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/coercions.py", line 573, in _literal_coercion
    element = list(element)
  File "/home/airflow/.local/lib/python3.8/site-packages/jinja2/runtime.py", line 852, in _fail_with_undefined_error
    raise self._undefined_exception(self._undefined_message)
jinja2.exceptions.UndefinedError: 'task_prepare_query_to_postgres' is undefined
[2024-07-18T16:02:28.218+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=google_analytics_data_etl, task_id=task_load_data_postgres, execution_date=20240718T150445, start_date=20240718T160227, end_date=20240718T160228
[2024-07-18T16:02:28.232+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 190 for task task_load_data_postgres ('task_prepare_query_to_postgres' is undefined; 1828)
[2024-07-18T16:02:28.271+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-07-18T16:02:28.325+0000] {taskinstance.py:3281} INFO - 0 downstream tasks scheduled from follow-on schedule check
